### PR TITLE
fix: link to bodacc

### LINKED ISSUE #86: Erreur sur un lien dans l'onglet annonces
Bonjour, je viens de tomber sur un lien qui ne marche pas. Ca semble reproductible sur toutes les entreprises.

Exemple : 
1) aller sur https://annuaire-entreprises.data.gouv.fr/annonces/306138900
2) cliquer sur le lien ` la page de cette entreprise`
3) le lien https://www.bodacc.fr/annonce/liste//306138900 a un double `/`

J'ai été tenté par une PR, mais je ne maitrise pas la stack utilisé, et j'ai abandonné la recherche de trouver le template correspondant :)

### PR DESCRIPTION
**Pull request by rmonnier9**
- Correction d'un bug.
- Détails :
  - Fix du lien
  - Refacto every replace with functions

Closes #86

### CODE REVIEW

File: app/(header-default)/annonces/[slug]/_components/bodacc.tsx [hunk 1]
```diff
@@ -68,7 +68,10 @@ const AnnoncesBodacc: React.FC<{
                 consulter{' '}
                 <UniteLegalePageLink
                   uniteLegale={uniteLegale}
-                  href={`${routes.bodacc.site.recherche}/${uniteLegale.siren}`}
+                  href={routes.bodacc.site.rechercheBySiren.replace(
+                    '{siren}',
+                    uniteLegale.siren
+                  )}
```
Comment (XavierJp) [inline app/(header-default)/annonces/[slug]/_components/bodacc.tsx:74]: Would prefer an url without the replace when possible

**Review by XavierJp — CHANGES_REQUESTED** [review]

**Review by XavierJp — COMMENTED** [review]
Sounds good to me. Only tiny issue I see is that you mutate route.ts signature. Now it can either be a function or a string. Why not only have string ?

**Review by XavierJp — APPROVED** [review]

**Comment by rmonnier9** [comment]
> Sounds good to me. Only tiny issue I see is that you mutate route.ts signature. Now it can either be a function or a string. Why not only have string ?

If we want to keep only strings in routes.ts, I don’t see another way than using placeholders and replacing them with .replace(). We could create a utility function to handle this for us, but it might make it less transparent what’s happening under the hood.

**Comment by rmonnier9** [comment]
I'm actually ok with both options

**Comment by XavierJp** [comment]
I like your current solution

**Pull request merged**